sdk/spring/spring-reference.yml

```diff
@@ -1,42 +1,55 @@
-- name: Azure Spring Boot Bom
-  content:
-    - name: Azure Spring Boot Bom
-      id: spring-boot-bom
-      description: BOM for Microsoft Azure Spring Boot support.
-      artifacts:
+# This file is aimed to build package reference of Azure Spring libraries to support an Intellij plugin for Azure SDK reference.
+#
+#The yaml follows structure of Spring Initializr's library mapping schema(https://github.com/spring-io/start.spring.io/blob/master/start-site/src/main/resources/application.yml) and thus is compatible of keys under property of `dependencies` to describe a library.
+#To make things clear, we call sdk/spring/spring-reference.yml as Spring yaml, and Spring Initializr's library mapping schema as template yaml.
+#For those keys not listed in Spring yaml, we can add them in case of future usage.
+#
+#Hierarchy of Spring yaml is organized as : Service -> Feature -> Package. Taking Cosmos DB starter as example, it is described as Cosmos DB -> Spring Data Cosmos -> azure-spring-boot-starter-cosmos.
+
+- name: Azure Spring Boot Bom                                                                       # Name of Azure service which the spring library relies on.
+                                                                                                    #    - For Spring library of which the underlying package is covered by [java-packages.csv](https://github.com/Azure/azure-sdk/blob/master/_data/releases/latest/java-packages.csv), we use its underlying library's `ServiceName`as the service name.
+                                                                                                    #    - For the rest like AAD starter, we create the service name ourselves, and try to follow the specification of java-pacakges.csv.
+  content:                                                                                          # Features under an Azure Service.
+    - name: Azure Spring Boot Bom                                                                   # Name of a feature under the associated Azure service which the spring library relies on.
+                                                                                                    #    - For Spring library of which the underlying package is covered by [java-packages.csv](https://github.com/Azure/azure-sdk/blob/master/_data/releases/latest/java-packages.csv), we use its underlying library's `DisplayName`as the feature name.
+                                                                                                    #    - For the rest, we create the feature name ourselves, and try to follow the specification of java-pacakges.csv.
+      id: spring-boot-bom                                                                           # Id of feature name that is not used now and reserved in case of future usage for reference.
```

Comment:
`Id of feature name that is not used ...` -> `Id of feature name. It is not used ...`